product_multi_company/models/product_category.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2015-2016 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import models, fields
+
+
+class ProductCategory(models.Model):
+    _inherit = "product.category"
+
+    def get_user_companies(self):
+        return self.env.user.company_ids.ids
+
+    total_route_ids = fields.Many2many(
+        domain=[
+            "|",
+            ("company_id", "=", False),
+            ("company_id", "in", "get_user_companies"),
+        ]
```

Comment:
I think the proper way would be to create a new field computed field `allowed_route_ids` and then update this domain to be: `domain="[('id', 'in', allowed_route_ids)]

Reply:
Tried that but got recursion error as long as allowed_route_ids needed route_ids in order to calculate those :( Now I think it is working, but still checking